### DISCUSSION TITLE
I can't use the templates at all: Templates folder is not defined or does not exist.

### DISCUSSION THREAD

**Discussion by xjohnyknox** [epwalsh/obsidian.nvim#691] [Q&A]
I keep getting the same error when I try to use the create from template option.
`:ObsidianTemplate`

<img width="631" alt="image" src="https://github.com/user-attachments/assets/f6addc91-e45e-47b3-b5c6-62499cd76785">

This is what I have configured so far on my plugin.lua config:

```
  -- Optional, for templates (see below).
    templates = {
      folder = "templates",
      date_format = "%Y-%m-%d",
      time_format = "%H:%M",
      -- A map for custom variables, the key should be the variable and the value a function
      substitutions = {},
  },
```
And the folder already exists:
```
❯ pwd
/Users/xjohnyx/Library/CloudStorage/SeaDrive-Johny/My Libraries/Obsidian/2ndbrain
❯ ls
1724277073-VPGP 2.md 1724279328-HOMA.md   Kindle-Highlights    Professional
1724277073-VPGP.md   Excalidraw           Personal             Templates

```

Any ideas whats missing?

Appreciate it.

**Comment by r2d2rogers** [comment]
Your file system may be case sensitive, check  if changing to `folder = "Templates",` sorts out the error.

**Comment by dc-jbw** [comment]
My system is case sensitive, and I have the same problem. No matter how carefully I specify the templates folder, I get that same error about how it's not defined or doesn't exist.

**Comment by jgrove2** [comment]
Hey has anyone found a way around this I am facing the same issue on my machine?

**Reply by dc-jbw** [reply]
FWIW, I stopped having the problem. I'm honestly not sure why, but as long as I specify the _case-sensitive_ name of the templates folder, and relative to the root of the vault I should also note, it works just fine.

**Reply by trasferetti** [reply]
I'm having the same problem. Changing the case didn't help, or the path (absolute or relative). My "templates" folder sits next to my "personal" and "work" folders, which are working fine.

**Reply by utilitron000** [reply]
I am also having this problem, would love to hear about anyone's solutions!

**Reply by r2d2rogers** [reply]
If you can post a gist link to your setup I would be glad to have a look.

I'm on Linux, but also use WSL on windows.

**Comment by utilitron000** [comment]
Thanks so very much for reaching out about this; I was actually able to resolve it -- I am a n00b and put the templates folder definition in the wrong section of the config. I'll stop spamming the github issues now!

**Comment by biscotty666** [comment]
I am having this problem as well. Some seem to have solved it but I don't see a solution. Thank you.

**Reply by Westo48** [reply]
https://www.youtube.com/watch?v=b-TJYCrYOWQ

This solved it for me in case you're still having issues with it

**Comment by Westo48** [comment]
~~Multiple have said they solved it, but didn't specify WHAT solved it. Any solutions??~~

This [YouTube video](https://www.youtube.com/watch?v=b-TJYCrYOWQ) solved it for me. Hope this helps someone in the future :)

**Reply by biscotty666** [reply]
It doesn't seem to mention obsidian.nvim, just a general intro to templates. Not sure how that's meant to help.

**Reply by Westo48** [reply]
It shows exactly how to set up the template... It goes step by step. Create a new folder, then go to the templates settings and type in the name of that folder.

The reason I was getting the same error as everyone else is I was trying to use file explorer and then use the folder address to link that folder to Obsidian. Secondly, I was using a folder outside the vault's folder. If that doesn't work for you then I wish you the best of luck, but that fix worked for me so I thought I'd share.

**Reply by biscotty666** [reply]
Templates work fine in Obsidian. The problem is with obsidian.nvim.

**Comment by biscotty666** [comment]
As a work-around, since we have obsidian-cli now (!), the commands `r!obsidian open file="file_name"` and `r!obsidian template:insert name="template-name"` can be used.
